internal/pkg/process/info.go

```diff
@@ -152,3 +170,38 @@ func (i *Info) GetFunctionReturns(name string) ([]uint64, error) {
 
 	return nil, fmt.Errorf("could not find returns for function %s", name)
 }
+
+// onceResult is an object that will perform exactly one action if that action
```

Comment:
nit: maybe rename it to something like `eventualResult` just to indicate that it can actually run more than once?

Reply:
Based on https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/1945#discussion_r2006130255, given this is going to only be used for a single type, I'm going to rename to `allocOnce`.

Reply:
Updated. PTAL.